src/prefect/environments/storage/gcs.py

```diff
@@ -0,0 +1,143 @@
+                a file path where a Flow has been serialized to
+
+        Returns:
+            - Flow: the requested flow
+
+        Raises:
+            - ValueError: if the flow is not contained in this storage
+        """
+        if not flow_location in self.flows.values():
+            raise ValueError("Flow is not contained in this Storage")
+
+        bucket = self._gcs_client.get_bucket(self.bucket)
+
+        self.logger.info("Downloading {} from {}".format(flow_location, self.bucket))
+
+        blob = bucket.get_blob(flow_location)
+        if not blob:
+            raise StorageError(
+                f"Flow not found in bucket: flow={flow_location} bucket={self.bucket}"
+            )
+        content = blob.download_as_string()
+        return cloudpickle.loads(content)
+
+    def add_flow(self, flow: "Flow") -> str:
+        """
+        Method for storing a new flow as bytes in a GCS bucket.
+
+        Args:
+            - flow (Flow): a Prefect Flow to add
+
+        Returns:
+            - str: the key of the newly added flow in the GCS bucket
+
+        Raises:
+            - ValueError: if a flow with the same name is already contained in this storage
+        """
+
+        if flow.name in self:
+            raise ValueError(
+                'Name conflict: Flow with the name "{}" is already present in this storage.'.format(
+                    flow.name
+                )
+            )
+
+        content = cloudpickle.dumps(flow)
+
+        # create key for Flow that uniquely identifies Flow object in GCS
+        key = self.key or "{}/{}".format(
+            slugify(flow.name), slugify(pendulum.now("utc").isoformat())
+        )
+
+        self.flows[flow.name] = key
+        self._flows[flow.name] = flow
+        return key
+
+    def __contains__(self, obj: Any) -> bool:
+        """
+        Method for determining whether an object is contained within this storage.
+        """
+        if not isinstance(obj, str):
+            return False
+        return obj in self.flows
+
+    def build(self) -> "Storage":
+        """
+        Build the GCS storage object by uploading Flows to an GCS bucket. This will upload
+        all of the flows found in `storage.flows`.
+        Returns:
+            - Storage: an GCS object that contains information about how and where
+                each flow is stored
+        """
+        for flow_name, flow in self._flows.items():
+            content = cloudpickle.dumps(flow)
+
+            bucket = self._gcs_client.get_bucket(self.bucket)
+
+            blob = bucket.blob(blob_name=self.flows[flow_name])
+
+            self.logger.info(
+                "Uploading {} to {}".format(self.flows[flow_name], self.bucket)
+            )
+
+            blob.upload_from_string(content)
+
+        return self
+
+    @property
+    def _gcs_client(self):  # type: ignore
+        from google.cloud import storage
+
+        return storage.Client(project=self.project)
```

Comment:
How are GCP creds handled with this? Does it require the user to log in outside of running this, creds in a known location, etc.?

Reply:
indeed, this requires the user to already have authenticated with the `gcloud` cli and make the credentials path available via the environment variable